src/FileProviders/Manifest.MSBuildTask/src/GenerateEmbeddedResourcesManifest.cs

```diff
@@ -90,11 +90,11 @@ public Manifest BuildManifest(EmbeddedItem[] processedItems)
         return manifest;
     }
 
-    private static string GetManifestPath(ITaskItem taskItem) => string.Equals(taskItem.GetMetadata(LogicalName), taskItem.GetMetadata(ManifestResourceName)) ?
+    private static string GetManifestPath(ITaskItem taskItem) => string.IsNullOrEmpty(taskItem.GetMetadata(LogicalName)) || string.Equals(taskItem.GetMetadata(LogicalName), taskItem.GetMetadata(ManifestResourceName)) ?
         taskItem.GetMetadata(TargetPath) :
         NormalizePath(taskItem.GetMetadata(LogicalName));
 
-    private static string GetAssemblyResourceName(ITaskItem taskItem) => string.Equals(taskItem.GetMetadata(LogicalName), taskItem.GetMetadata(ManifestResourceName)) ?
+    private static string GetAssemblyResourceName(ITaskItem taskItem) => string.IsNullOrEmpty(taskItem.GetMetadata(LogicalName)) || !string.IsNullOrEmpty(taskItem.GetMetadata(LogicalName)) && string.Equals(taskItem.GetMetadata(LogicalName), taskItem.GetMetadata(ManifestResourceName)) ?
```

Comment:
I think we can remove the "`!string.IsNullOrEmpty(taskItem.GetMetadata(LogicalName))`" check, since that expression will only get reached if the previous "`string.IsNullOrEmpty(taskItem.GetMetadata(LogicalName))`" check evaluates to `false`.

Reply:
🤦 Thanks for catching that! Was a remnant of when I was initially floundering and missed it 'cause it ran off the screen. My bad!

Fixed!